beacon-chain/rpc/lookup/blocker.go

```diff
@@ -461,3 +462,156 @@ func (p *BeaconDbBlocker) neededDataColumnSidecars(root [fieldparams.RootLength]
 
 	return verifiedRoSidecars, nil
 }
+
+// DataColumns returns the data column sidecars for a given block id identifier and column indices. The identifier can be one of:
+//   - "head" (canonical head in node's view)
+//   - "genesis"
+//   - "finalized"
+//   - "justified"
+//   - <slot>
+//   - <hex encoded block root with '0x' prefix>
+//   - <block root>
+//
+// cases:
+//   - no block, 404
+//   - block exists, before Fulu fork, 400 (data columns are not supported before Fulu fork)
+//   - block exists, has commitments, inside retention period serve them w/ 200 unless we hit an error reading them.
+//     we are technically not supposed to import a block to forkchoice unless we have the data columns, so the nuance here is if we can't find the file and we are inside the protocol-defined retention period, then it's actually a 500.
+//   - block exists, has commitments, outside retention period - ie just like block exists, no commitment
```

Comment:
Because this is a debug endpoint, shouldn't we just return whatever we have? You changed the blobs endpoint this way too, even though it's not debug: https://github.com/OffchainLabs/prysm/pull/15610#discussion_r2353504518

Actually I don't see any code which checks the retention period, but there is a test case for this...

Reply:
we should remove this from both blobs and this endpoint

Reply:
well in this case it's just a utility service but should be removed from both